oracle/types/provider.go

```diff
@@ -82,5 +83,7 @@ func (t *ProviderTickers) Add(ticker ProviderTicker) {
 	t.mut.Lock()
 	defer t.mut.Unlock()
 
+	t.cache[strings.ToLower(ticker.GetOffChainTicker())] = ticker
 	t.cache[ticker.GetOffChainTicker()] = ticker
+	t.cache[strings.ToUpper(ticker.GetOffChainTicker())] = ticker
```

Comment:
Adding this since some times the subscription is lower case but response is upper case. Should be a safe operation.

Reply:
Why do we need all 3 of these instead of using strictly upper or strictly lower?

Reply:
brevity mostly, ideally we just use some type of regex.

Reply:
I don't think this makes sense. Ideally something like this works:

Adding to cache:
```
t.cache[strings.ToUpper(ticker.GetOffChainTicker())] = ticker
```
Retrieving from cache:
```
return t.cache[strings.ToUpper(ticker.GetOffChainTicker())]
```

Why do we need a regex ever?

Reply:
some providers dont specify whether we retrieve responses in the same casing as the request, which could lead to total liveness failure on a given provider. regex is probably overkill but if the offchain ticker happens to be upper case and responses are lower case we would effectively not be able to parse out the responses.

Reply:
We could change the API to the `ProviderTickers` func but that would be a much larger change.